### PR TITLE
Task/day 3 frontend

### PR DESCRIPTION
**Pull request by BrianZunigaEncora**
Frontend screens added with partial functionability
image

image

### CODE REVIEW

File: productsManagement/src/main/webapp/WEB-INF/views/add-product.jsp [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+            <div class="alert alert-danger" role="alert">Error adding Product: ${savedProduct.name}</div>
+        </div>
+    </div>
+</c:if>
+
+<c:url var="add_product_url" value="/products/add"/>
+<div class="row">
+    <div class="col-md-4 col-md-offset-4">
+        <h2>Add Product</h2>
+        <form:form action="${add_product_url}" method="post" modelAttribute="product">
```
Comment (juan-caravantes) [inline productsManagement/src/main/webapp/WEB-INF/views/add-product.jsp:35]: Don't forget validations.

File: commons/src/main/java/com/skytouch/entity/ProductEntity.java [hunk 2]
```diff
@@ -1,16 +1,25 @@
...
 
 import java.math.BigDecimal;
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.Id;
 
-@Data
-@AllArgsConstructor
-@NoArgsConstructor
+@Entity
```
Comment (juan-caravantes) [inline commons/src/main/java/com/skytouch/entity/ProductEntity.java:12]: Why do you need `@Entity` annotation if you have to use stored procedures?

File: productsManagement/src/main/webapp/WEB-INF/views/index.jsp
Comment (juan-caravantes) [inline productsManagement/src/main/webapp/WEB-INF/views/index.jsp:1]: What about error page?

File: productsManagement/src/main/java/com/productsmanagement/controller/ProductManagementController.java [hunk 3]
```diff
@@ -1,13 +1,18 @@
...
 import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.bind.annotation.PostMapping;
 import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.servlet.mvc.support.RedirectAttributes;
+import org.springframework.web.servlet.view.RedirectView;
 
 @Controller
-@RequestMapping("/product-management")
+@RequestMapping("")
```
Comment (AlexmuosoEncora) [inline productsManagement/src/main/java/com/productsmanagement/controller/ProductManagementController.java:15]: is it needed ?

File: productsManagement/src/main/java/com/productsmanagement/util/Utils.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
+package com.productsmanagement.util;
+
+import com.skytouch.dto.ProductDTO;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class Utils {
```
Comment (AlexmuosoEncora) [inline productsManagement/src/main/java/com/productsmanagement/util/Utils.java:10]: have u think on use @NoArgsConstructor(access = AccessLevel.PRIVATE)  ?

File: productsManagement/src/main/java/com/productsmanagement/controller/ProductManagementController.java [hunk 5]
```diff
@@ -16,11 +21,30 @@ public class ProductManagementController {
...
+    public String addBookView(Model model) {
+        model.addAttribute("product", new ProductDTO());
+        return "add-product";
+    }
+
+    @PostMapping("/products/add")
+    public RedirectView addProductAction(@ModelAttribute("product") ProductDTO product, RedirectAttributes redirectAttributes) {
+        final RedirectView redirectView = new RedirectView("/products/add", true);
+        ProductDTO savedProduct = productManagementService.addProduct(product);
+        redirectAttributes.addFlashAttribute("savedProduct", savedProduct);
```
Comment (AlexmuosoEncora) [inline productsManagement/src/main/java/com/productsmanagement/controller/ProductManagementController.java:45]: addFlashAttribute or addAttribute ?

**Pull request merged**